pkgs/misc/vim-plugins/overrides.nix

```diff
@@ -267,6 +268,44 @@ self: super: {
     dependencies = [ self.fzfWrapper ];
   });
 
+  markdown-preview = let
+    src = fetchFromGitHub {
+      owner = "iamcco";
+      repo = "markdown-preview.nvim";
+      rev = "d319eaac9ef155d2e2cb846c6754a22e9e8a494a";
+      sha256 = "1mch9s2fbk068q2hrxz5ksrw4b15h8vdxg1zi9ads9s4v8rb3hcg";
+    };
+    app = mkYarnPackage {
+      name = "markdown-preview-app";
+      src = "${src}/app";
+      packageJSON = ./plugins/markdown-preview/package.json;
+      yarnLock = ./plugins/markdown-preview/yarn.lock;
+      yarnNix = ./plugins/markdown-preview/yarn.nix;
```

Comment:
I don't think that you need to add separate `package.json` and `yarn.lock` files to this repo. They are both included in the `markdown-preview.nvim` repo. I think that you just need to do something like:

```
packageJSON = src + "/package.json";
yarnLock = src + "/package.json";
```

And since the `yarn.lock` should ensure stable results, you shouldn't need to check in a `yarn.nix` with this PR; `yarn2nix` should be able to generate one itself.

At least this is my understanding. I could definitely be wrong.

Reply:
Unfortunately the _ofborg_ checks in this PR fail when any of the 3 files are not local to the `nixpkgs` repo.

Reply:
I thought that the original problem was that it was trying to look in `app/package.json` because of its default relative to the `src` that was defined as `src + "/app"`. Because of the squashing it's tough for me to see the history of what failed. Was there a commit where you fixed that but it still complained that the files weren't local?

Reply:
Yes, I had incrementally added those files and watched _ofborg_ fail. At @jonringer’s recommendation I squashed the commits into 1 so all that history is lost.

In reality all 3 files can be omitted and the plugin builds fine. It’s only these _ofborg_ tests that fail. If you look at other usages of `mkYarnPackage`  in `nixpkgs` you will notice that they have these 3 files commited:
https://github.com/NixOS/nixpkgs/tree/75ee18766ac52a97df1dc55a11a2d44cb0e68d0e/pkgs/servers/web-apps/codimd

Reply:
This is mostly likely due to "Import from derivation" or IFD. Ofborg doesn't like you creating one derivation, then re-using outputs of that derivation to create another. Which is why you need to check-in the deps file, which eliminates one of the links in the import chain.

Reply:
@oxalica you've done some node work.

Reply:
Why can't we put this under nodePackages?